pyproject.toml

```diff
@@ -4,6 +4,9 @@ requires = [
     "wheel",
     # On Windows, use the CasADi vcpkg registry and CMake bundled from MSVC
     "casadi>=3.6.3; platform_system!='Windows'",
+    # Note: the version of CasADi as a build-time dependency should be matched
```

Comment:
I haven't bumped the minimum version bound for `casadi` in `pyproject.toml` for now because conda-forge still has the latest release as `3.6.3`. We can still use `3.6.4` for linkage, because when building the wheels for Linux and macOS, `3.6.4` is downloaded from `pip` (the same as the version bumped to in the registry). It is the same case for `pip` installations. We need to ensure that `3.6.X` is used for the linkage (i.e., there haven't been issues between patch versions w.r.t serialisation) and the registry is updated periodically going forward.